prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirTransformer.kt

```diff
@@ -0,0 +1,225 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7
+
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.FHIRTransformSchemaElement
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.FhirTransformSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.fhirTransformSchemaFromFile
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.ConstantSubstitutor
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.CustomContext
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirBundleUtils
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirPathUtils
+import org.apache.commons.io.FilenameUtils
+import org.apache.logging.log4j.Level
+import org.hl7.fhir.r4.model.Base
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.Extension
+
+/**
+ * Transform a FHIR bundle based on the [schemaRef].
+ * The transformer will error out if [strict] is set to true and there is an error during the translation. If [strict]
+ * is set to false (the default) then any translation errors are logged as a warning. Note [strict] does not affect
+ * the schema validation process.
+ * @property constantSubstitutor the constant substitutor. Should be a static instance, but is not thread safe
+ */
+class FhirTransformer(
+    private val schemaRef: FhirTransformSchema,
+    private val strict: Boolean = false,
+    // the constant substitutor is not thread safe, so we need one instance per converter instead of using a shared copy
+    private val constantSubstitutor: ConstantSubstitutor = ConstantSubstitutor()
+) : ConfigSchemaProcessor() {
+    /**
+     * Transform a FHIR bundle based on the [schema] in the [schemaFolder] location.
+     * The transformer will error out if [strict] is set to true and there is an error during the translation. If [strict]
+     * is set to false (the default) then any translation errors are logged as a warning. Note [strict] does not affect
+     * the schema validation process.
+     */
+    constructor(
+        schema: String,
+        schemaFolder: String,
+        strict: Boolean = false,
+    ) : this(
+        schemaRef = fhirTransformSchemaFromFile(schema, schemaFolder),
+        strict = strict,
+    )
+
+    /**
+     * Transform a FHIR bundle based on the [schema] (which includes a folder location).
+     * The transformer will error out if [strict] is set to true and there is an error during the translation. If [strict]
+     * is set to false (the default) then any translation errors are logged as a warning. Note [strict] does not affect
+     * the schema validation process.
+     */
+    constructor(
+        schema: String,
+        strict: Boolean = false,
+    ) : this(
+        schemaRef = fhirTransformSchemaFromFile(
+            FilenameUtils.getName(schema),
+            FilenameUtils.getPathNoEndSeparator(schema)
+        ),
+        strict = strict,
+    )
```

Comment:
IntelliJ complains that these constructors are never used. Are they needed?

Reply:
I suspect at least some of them will be eventually when this is integrated more into the pipeline, for now this is still only used in tests, so the constructors aren't as needed. After that integration point in a future story, I'll remove any that are still unused.